client/app.jsx

```diff
@@ -23,17 +24,28 @@ export default class App extends React.Component {
     if (route.path === '') {
       return <StartPage />;
     }
+    if (route.path === 'sign-up') {
+      return <AuthPage />;
+    }
     if (route.path === 'workouts') {
       const workoutId = route.params.get('workoutId');
       return <AddExercisePage workoutId={workoutId} />;
     }
   }
 
   render() {
+    const { route } = this.state;
+    const contextValue = { route };
     return (
-      <>
-      { this.renderPage() }
-      </>
+      <AppContext.Provider value={contextValue}>
+        <>
+        { this.renderPage() }
+        </>
+      </AppContext.Provider>
     );
   }
 }
+
+App.contextType = AppContext;
+AddExercisePage.contextType = AppContext;
+StartPage.contextType = AppContext;
```

Comment:
App does not need a contextType and it's better to add these properties in the modules that need them instead of all right here